src/EditorFeatures/CSharpTest/Completion/CompletionProviders/PartialTypeCompletionProviderTests.cs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders
+{
+    public class PartialTypeCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        public PartialTypeCompletionProviderTests(CSharpTestWorkspaceFixture workspaceFixture) : base(workspaceFixture)
+        {
+        }
+
+        internal override CompletionProvider CreateCompletionProvider()
+        {
+            return new PartialTypeCompletionProvider();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestRecommendTypesWithoutPartial()
+        {
+            var text = @"
+class C { }
+
+partial class $$";
+
+            await VerifyItemIsAbsentAsync(text, "C");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestPartialClass1()
+        {
+            var text = @"
+partial class C { }
+
+partial class $$";
+
+            await VerifyItemExistsAsync(text, "C");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestPartialGenericClass1()
+        {
+            var text = @"
+class Bar { }
+
+partial class C<Bar> { }
+
+partial class $$";
+
+            await VerifyItemExistsAsync(text, "C<Bar>");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestPartialGenericClassCommitOnTab()
+        {
+            var text = @"
+class Bar { }
+
+partial class C<Bar> { }
+
+partial class $$";
+
+            var expected = @"
+class Bar { }
+
+partial class C<Bar> { }
+
+partial class C<Bar>";
+
+            await VerifyProviderCommitAsync(text, "C<Bar>", expected, null, "");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestPartialGenericClassCommitOnSpace()
+        {
+            var text = @"
+partial class C<T> { }
+
+partial class $$";
+
+            var expected = @"
+partial class C<T> { }
+
+partial class C<T> ";
+
+            await VerifyProviderCommitAsync(text, "C<T>", expected, ' ', "");
```

Comment:
I think this new behavior is ok.  Because it's not explicitly selected, the user has to go and select that item explicitly.  Selecting it explicitly and then typing '<' is somewhat strange as you'd be effectively saying "i want completion's help in typing the type name... except for the generics, which i will type myself"

I more expect people to go and select the item in the list and then type "space, tab, enter or colon".  All of which will have the right behavior.  So i think how this is working now is ok.  What do you think Kevin?